projects/igniteui-angular/src/lib/grids/grid/grid.pinning.spec.ts

```diff
@@ -1,1140 +1,681 @@
+            it('should allow pinning/unpinning via the grid API', () => {
+
+                // Unpin column
+                grid.unpinColumn('CompanyName');
+                fix.detectChanges();
+
+                // verify column is unpinned
+                expect(grid.pinnedColumns.length).toEqual(1);
+                expect(grid.unpinnedColumns.length).toEqual(10);
+
+                const col = grid.getColumnByName('CompanyName');
+                expect(col.pinned).toBe(false);
+                expect(col.visibleIndex).toEqual(1);
+
+                // verify DOM
+                let cell = grid.getCellByColumn(0, 'CompanyName');
+                expect(cell.visibleColumnIndex).toEqual(1);
+                expect(GridFunctions.isCellPinned(cell)).toBe(false);
+
+                const secondHeader = GridFunctions.getColumnHeaders(fix)[1];
+
+                expect(secondHeader.context.column.field).toEqual('CompanyName');
+                expect(GridFunctions.isHeaderPinned(secondHeader)).toBe(false);
+
+                // verify container widths
+                expect(grid.pinnedWidth).toEqual(200);
+                expect(grid.unpinnedWidth + grid.scrollWidth).toEqual(600);
+
+                // pin back the column.
+                grid.pinColumn('CompanyName');
+                fix.detectChanges();
+
+                // verify column is pinned
+                expect(grid.pinnedColumns.length).toEqual(2);
+                expect(grid.unpinnedColumns.length).toEqual(9);
+
+                // verify container widths
+                expect(grid.pinnedWidth).toEqual(400);
+                expect(grid.unpinnedWidth + grid.scrollWidth).toEqual(400);
+
+                expect(col.pinned).toBe(true);
+                expect(col.visibleIndex).toEqual(grid.unpinnedColumns.length + 1);
+
+                cell = grid.getCellByColumn(0, 'CompanyName');
+                expect(cell.visibleColumnIndex).toEqual(grid.unpinnedColumns.length + 1);
+                expect(GridFunctions.isCellPinned(cell)).toBe(true);
+            });
+
+            it('should correctly pin column to right when row selectors are enabled.', () => {
+                grid.rowSelectable = true;
+                fix.detectChanges();
+
+                // check row DOM
+                const row = grid.getRowByIndex(0).nativeElement;
+
+                GridSelectionFunctions.verifyRowHasCheckbox(row);
+                expect(GridFunctions.getRowDisplayContainer(fix, 0)).toBeDefined();
+                expect(row.children[2].getAttribute('aria-describedby')).toBe(grid.id + '_CompanyName');
+                expect(row.children[3].getAttribute('aria-describedby')).toBe(grid.id + '_ContactName');
+
+                // check scrollbar DOM
+                const scrBarStartSection = fix.debugElement.query(By.css('.igx-grid__scroll-start'));
+                const scrBarMainSection = fix.debugElement.query(By.css('.igx-grid__scroll-main'));
```

Comment:
Constants for the classes should be added

Reply:
Done